### PR TITLE
Adding inline docs to base profile view

### PR DESCRIPTION
**Pull request by etoledom**
Closes #

### Description

Adding inline docs to base profile view.

### Testing Steps

- Build Docs
- Search for `BaseProfileView`.
- Check that all reads good.

### CODE REVIEW

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 1]
```diff
@@ -149,20 +162,24 @@ open class BaseProfileView: UIView, UIContentView {
         return label
     }()
 
+    /// The lable which displays the profile name text.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// The label which displays the profile name text.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 2]
```diff
@@ -43,6 +48,7 @@ open class BaseProfileView: UIView, UIContentView {
         }
     }
 
+    /// Wether the view is in loading state.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// Whether the view is in loading state.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 3]
```diff
@@ -149,20 +162,24 @@ open class BaseProfileView: UIView, UIContentView {
...
 
+    /// The lable which displays the profile name text.
     public private(set) lazy var displayNameLabel: UILabel = {
         let label = UILabel()
         label.translatesAutoresizingMaskIntoConstraints = false
         label.adjustsFontForContentSizeCategory = true
         return label
     }()
 
+    /// The lable which displays the profile personal information text,
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// The label which displays the profile personal information text,
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 4]
```diff
@@ -234,6 +261,14 @@ open class BaseProfileView: UIView, UIContentView {
         fatalError("init(coder:) has not been implemented")
     }
 
+    /// Loads an avatar to be displayed in the avatar view.
+    /// - Parameters:
+    ///   - avatarIdentifier: The identifier of the avatar to be loaded.
+    ///   - placeholder: An image which will be temporarely displayed while the avatar loads.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    ///   - placeholder: An image which will be temporarily displayed while the avatar loads.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 5]
```diff
@@ -416,95 +453,28 @@ open class BaseProfileView: UIView, UIContentView {
     }
 }
 
+/// Methods for managing actions on the profile view.
 public protocol ProfileViewDelegate: NSObjectProtocol {
+    /// Tells the delegate the profile action button of the view has been tapped.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// Tells the delegate that the profile action button of the view has been tapped.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 6]
```diff
@@ -416,95 +453,28 @@ open class BaseProfileView: UIView, UIContentView {
...
+    ///
+    /// The `profileURL` contains a url which corresponds to the button intended action declared by the button style.
+    /// > The expected action to this delegate method call is to present this URL in an internal Safari view.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// > The recommended action for this delegate method is to present this URL in a `SFSafariViewController`.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 7]
```diff
@@ -416,95 +453,28 @@ open class BaseProfileView: UIView, UIContentView {
...
     func profileView(_ view: BaseProfileView, didTapOnProfileButtonWithStyle style: ProfileButtonStyle, profileURL: URL?)
+    /// Tells the delegate one of the profile associated accounts button has been tapped.
+    ///
+    /// The `accountModel` contains the information about the profile's account which was tapped, including a url to the account on that service.
+    /// > The expected action to this delegate method call is to present this account URL in an internal Safari view.
+    /// - Parameters:
+    ///   - view: The profile view informing about the tap.
+    ///   - accountModel: Information about the associated account which was tapped.
     func profileView(_ view: BaseProfileView, didTapOnAccountButtonWithModel accountModel: AccountModel)
+    /// Tells the delegate the profile avatar view has been tapped.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// Tells the delegate that the profile avatar view has been tapped.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 8]
```diff
@@ -416,95 +453,28 @@ open class BaseProfileView: UIView, UIContentView {
...
+    /// The `profileURL` contains a url which corresponds to the button intended action declared by the button style.
+    /// > The expected action to this delegate method call is to present this URL in an internal Safari view.
+    /// - Parameters:
+    ///   - view: The profile view informing about the tap.
+    ///   - style: The current profile action button style.
+    ///   - profileURL: A possible URL to be presented to the user
     func profileView(_ view: BaseProfileView, didTapOnProfileButtonWithStyle style: ProfileButtonStyle, profileURL: URL?)
+    /// Tells the delegate one of the profile associated accounts button has been tapped.
+    ///
+    /// The `accountModel` contains the information about the profile's account which was tapped, including a url to the account on that service.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// The `accountModel` contains information about the tapped account including a URL to it.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 9]
```diff
@@ -416,95 +453,28 @@ open class BaseProfileView: UIView, UIContentView {
...
+    /// > The expected action to this delegate method call is to present this account URL in an internal Safari view.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// > The recommended action for this delegate method is to present this URL in a `SFSafariViewController`.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 10]
```diff
@@ -181,12 +199,21 @@ open class BaseProfileView: UIView, UIContentView {
...
 
+    /// The palette used to set the profile view colors.
     public var paletteType: PaletteType {
         didSet {
             refresh(with: paletteType)
         }
     }
 
+    /// Creates an instance of BaseProfileView. **Do not** create an instance directly.
+    /// >  This class is intended as an abstract class. You can subclass BaseProfileView, and call this init method as `super.init(...)`
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// >  This class is intended as an abstract class. You can subclass BaseProfileView, and call this init method as `super.init(...)`. You should override `arrangeSubviews()` to put your views in the `rootStackView`.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 11]
```diff
@@ -106,6 +115,8 @@ open class BaseProfileView: UIView, UIContentView {
         }
     }
 
+    /// Empty stack view added to the view's hierarchy.
+    /// Use this stack as a base to create your own Profile View design.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    /// Use this stack as a base to create your own Profile View design. You should override `arrangeSubviews()` to put your views in the `rootStackView`.
```

(it needs to happen there for placeholders to display nicely.)

**Review by pinarol — APPROVED** [review]
Just some suggestions.

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 12]
```diff
@@ -416,95 +453,28 @@ open class BaseProfileView: UIView, UIContentView {
...
+    /// The `accountModel` contains the information about the profile's account which was tapped, including a url to the account on that service.
+    /// > The expected action to this delegate method call is to present this account URL in an internal Safari view.
+    /// - Parameters:
+    ///   - view: The profile view informing about the tap.
+    ///   - accountModel: Information about the associated account which was tapped.
     func profileView(_ view: BaseProfileView, didTapOnAccountButtonWithModel accountModel: AccountModel)
+    /// Tells the delegate the profile avatar view has been tapped.
+    /// - Parameters:
+    ///   - view: The profile view informing about the tap.
+    ///   - avatarID: The ID of the avatar tapped.
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift]:
```suggestion
    ///   - avatarID: The ID of the avatar tapped. Check out ``AvatarURL`` to create an avatar URL from an `AvatarIdentifier`.
```

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift (hunk 11 above)
Comment (etoledom) [reply Sources/GravatarUI/ProfileView/BaseProfileView.swift]: Good call! We can make a tutorial about this also

**Comment by etoledom** [comment]
Thank you for all the suggestions 🙏

**Pull request merged**